### PR TITLE
CLN: str.format -> f-strings for `io/sas`

### LINKED ISSUE #29547: Replace old string formatting syntax with f-strings
Since we no longer support python 3.5, we can now use the new f-strings instead of the old ```.format()``` ( and obviously the ```%``` formatting).

#### Notes:
* Don't forget to link this issue in your pull request's body message , simply paste this ```https://github.com/pandas-dev/pandas/issues/29547``` in your pull request's body message.

* If any of your changed files are related to #29886 , please make sure to link your pull request to that issue as well, simply paste this ```https://github.com/pandas-dev/pandas/issues/29886``` in your pull request's body message as well.

* Please comment what you are planning to work on, so we won't do double work.

* If a file/files that should be marked as done, is not marked, please comment letting me know.

---

To check what files still needs to be fixed in the ```pandas``` directory:
```
grep -l -R '%s'  --include=*.{py,pyx} pandas/
grep -l -R '%d' --include=*.{py,pyx} pandas/
grep -l -R '\.format(' --include=*.{py,pyx} pandas/
```

All of the above can also be used as a one liner:
```
grep -l -R -e '%s' -e '%d' -e '\.format(' --include=*.{py,pyx} pandas/
```

##### Tip:
If you want to see the line number of the occurrence, replace the ```-l``` with ```-n```
for example:
```
grep -n -R '%s' --include=*.{py,pyx} pandas/
```

---

### The current list is:

- [ ] pandas/compat/pickle_compat.py

- [ ] pandas/_config/config.py

- [ ] pandas/core/arrays/datetimelike.py

- [ ] pandas/core/arrays/datetimes.py

- [ ] pandas/core/arrays/integer.py

- [ ] pandas/core/arrays/period.py

- [ ] pandas/core/computation/pytables.py

- [ ] pandas/core/config_init.py

- [ ] pandas/core/frame.py

- [ ] pandas/core/generic.py

- [ ] pandas/core/groupby/generic.py

- [ ] pandas/core/groupby/groupby.py

- [ ] pandas/core/indexes/base.py

- [ ] pandas/core/indexes/multi.py

- [ ] pandas/core/indexes/range.py

- [ ] pandas/core/ops/docstrings.py

- [ ] pandas/core/ops/\_\_init__.py

- [ ] pandas/core/reshape/merge.py

- [ ] pandas/core/tools/datetimes.py

- [ ] pandas/io/formats/css.py

- [ ] pandas/io/formats/excel.py

- [ ] pandas/io/formats/format.py

- [ ] pandas/io/formats/html.py

- [ ] pandas/io/formats/info.py

- [ ] pandas/io/formats/latex.py

- [ ] pandas/io/formats/printing.py

- [ ] pandas/io/formats/style.py

- [ ] pandas/io/parsers.py

- [ ] pandas/io/pytables.py

- [ ] pandas/io/sas/sas_xport.py

- [ ] pandas/io/stata.py

- [ ] pandas/_libs/tslibs/c_timestamp.pyx

- [ ] pandas/_libs/tslibs/frequencies.pyx

- [ ] pandas/_libs/tslibs/parsing.pyx

- [ ] pandas/_libs/tslibs/period.pyx

- [ ] pandas/_libs/tslibs/strptime.pyx

- [ ] pandas/_libs/tslibs/timedeltas.pyx

- [ ] pandas/plotting/_matplotlib/converter.py

- [ ] pandas/tests/arrays/categorical/test_operators.py

- [ ] pandas/tests/arrays/test_datetimelike.py

- [ ] pandas/tests/dtypes/test_dtypes.py

- [ ] pandas/tests/extension/base/setitem.py

- [ ] pandas/tests/frame/test_constructors.py

- [ ] pandas/tests/frame/test_missing.py

- [ ] pandas/tests/frame/test_to_csv.py

- [ ] pandas/tests/groupby/aggregate/test_other.py

- [ ] pandas/tests/indexes/datetimes/test_date_range.py

- [ ] pandas/tests/indexes/datetimes/test_datetime.py

- [ ] pandas/tests/indexes/datetimes/test_formats.py

- [ ] pandas/tests/indexes/datetimes/test_partial_slicing.py

- [ ] pandas/tests/indexes/interval/test_constructors.py

- [ ] pandas/tests/indexes/interval/test_interval.py

- [ ] pandas/tests/indexes/multi/test_format.py

- [ ] pandas/tests/indexes/period/test_formats.py

- [ ] pandas/tests/indexes/test_base.py

- [ ] pandas/tests/indexes/timedeltas/test_timedelta.py

- [ ] pandas/tests/indexing/test_categorical.py

- [ ] pandas/tests/indexing/test_coercion.py

- [ ] pandas/tests/io/excel/test_openpyxl.py

- [ ] pandas/tests/io/excel/test_writers.py

- [ ] pandas/tests/io/formats/test_format.py

- [ ] pandas/tests/io/formats/test_printing.py

- [ ] pandas/tests/io/formats/test_style.py

- [ ] pandas/tests/io/formats/test_to_csv.py

- [ ] pandas/tests/io/formats/test_to_html.py

- [ ] pandas/tests/io/formats/test_to_latex.py

- [ ] pandas/tests/io/parser/test_compression.py

- [ ] pandas/tests/io/parser/test_encoding.py

- [ ] pandas/tests/io/parser/test_header.py

- [ ] pandas/tests/io/parser/test_parse_dates.py

- [ ] pandas/tests/io/parser/test_usecols.py

- [ ] pandas/tests/io/test_html.py

- [ ] pandas/tests/io/test_sql.py

- [ ] pandas/tests/io/test_stata.py

- [ ] pandas/tests/reductions/test_reductions.py

- [ ] pandas/tests/reshape/test_concat.py

- [ ] pandas/tests/reshape/test_melt.py

- [ ] pandas/tests/scalar/period/test_period.py

- [ ] pandas/tests/scalar/timedelta/test_timedelta.py

- [ ] pandas/tests/scalar/timestamp/test_constructors.py

- [ ] pandas/tests/series/indexing/test_numeric.py

- [ ] pandas/tests/series/indexing/test_take.py

- [ ] pandas/tests/series/indexing/test_where.py

- [ ] pandas/tests/series/methods/test_rename.py

- [ ] pandas/tests/series/test_api.py

- [ ] pandas/tests/series/test_constructors.py

- [ ] pandas/tests/series/test_datetime_values.py

- [ ] pandas/tests/series/test_repr.py

- [ ] pandas/tests/test_strings.py

- [ ] pandas/tests/tools/test_to_datetime.py

- [ ] pandas/tests/tseries/holiday/test_calendar.py

- [ ] pandas/tests/tseries/holiday/test_holiday.py

- [ ] pandas/tests/tslibs/test_parsing.py

- [ ] pandas/tests/util/test_assert_frame_equal.py

- [ ] pandas/tseries/frequencies.py

- [ ] pandas/util/_decorators.py

- [ ] pandas/util/_test_decorators.py

- [ ] pandas/util/_validators.py

- [ ] pandas/_version.py

---

#### NOTE:

The list may change as files are moved/renamed constantly.

---

Inhereted files and commands from this [PR](https://github.com/pandas-dev/pandas/issues/16130).

### PR DESCRIPTION
**Pull request by jyscao**
- [x] ref #29547 
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by gfyoung** [comment]
@jyscao: Thanks for the contribution, and welcome to `pandas` !  🚀 

LGTM.  Codecov failure can be ignored.

**Comment by MomIsBestFriend** [comment]
>     * [x]  closes #29547


Can you please change the word ```closes``` to ```ref``` ?

File: pandas/io/sas/sas_xport.py [hunk 1]
```diff
@@ -367,8 +367,8 @@ def _read_header(self):
             fl = field["field_length"]
             if field["ntype"] == "numeric" and ((fl < 2) or (fl > 8)):
                 self.close()
-                msg = "Floating field width {0} is not between 2 and 8."
-                raise TypeError(msg.format(fl))
+                msg = "f{Floating field width {fl} is not between 2 and 8.}"
```
Comment (MomIsBestFriend) [inline pandas/io/sas/sas_xport.py]: The ```f``` is inside the string :)

Comment (jyscao) [reply pandas/io/sas/sas_xport.py]: How very sloppy of me... Good eye, it has been fixed now.

File: pandas/io/sas/sas7bdat.py [hunk 2]
```diff
@@ -169,7 +169,7 @@ def _get_properties(self):
         if buf in const.encoding_names:
             self.file_encoding = const.encoding_names[buf]
         else:
-            self.file_encoding = "unknown (code={name!s})".format(name=buf)
+            self.file_encoding = f"unknown (code={str(buf)})"
```
Comment (WillAyd) [inline pandas/io/sas/sas7bdat.py]:
```suggestion
            self.file_encoding = f"unknown (code={buf})"
```

Comment (WillAyd) [reply pandas/io/sas/sas7bdat.py]: I think we've been cleaning these up elsewhere; shouldn't need the `str()` call

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/io/sas/sas7bdat.py (hunk 2 above)
Comment (jyscao) [reply pandas/io/sas/sas7bdat.py]: changed, thanks

**Comment by jbrockmendel** [comment]
restarted the travis build.  LGTM.

File: pandas/io/sas/sas7bdat.py [hunk 3]
```diff
@@ -457,12 +457,9 @@ def _process_columnsize_subheader(self, offset, length):
...
-                "Warning: column count mismatch ({p1} + {p2} != "
-                "{column_count})\n".format(
-                    p1=self.col_count_p1,
-                    p2=self.col_count_p2,
-                    column_count=self.column_count,
-                )
+                f"Warning: column count mismatch ({self.col_count_p1} + "
+                f"{self.col_count_p2} != "
+                f"{self.column_count})\n"
             )
```
Comment (jbrockmendel) [inline pandas/io/sas/sas7bdat.py]: for follow-up, this probably should use warnings rather than print

**Comment by jbrockmendel** [comment]
Thanks @jyscao

**Pull request merged**